### DISCUSSION TITLE
why no getParameters method in Definitions class?

### DISCUSSION THREAD

**Discussion by stardustman** [rabbitmq/hop#398] [Q&A] [answered]
I get definitions using [http api/definitions](http://localhost:15672/api/definitions), there is a` parameters` key
<img width="823" alt="image" src="https://github.com/rabbitmq/hop/assets/22998204/5f825224-594d-4c6d-a59d-8e6f897d2f27">

where I using [client.getDefinitions()](https://github.com/rabbitmq/hop/blob/main/src/main/java/com/rabbitmq/http/client/domain/Definitions.java#L44) I can't get parameters. why? how can I get the `parameters`?

**Comment by acogoluegnes** [comment]
Parameters are not available in the `Definitions` class. They don't have a fixed structure, so it may be awkward to map them, but there should be a reasonable way to do it. Would you mind sharing the commands to define the parameters you are using so I can make some tests? Thanks.

**Reply by michaelklishin** [reply]
The only reason has already been stated, runtime parameters are pretty free-form so at best we can return a generic collection… our Rust and Go HTTP API clients do just that IIRC.

**Reply by stardustman** [reply]
ok. Thanks.
my commands about set up federation upstream
```bash
rabbitmqctl set_parameter federation-upstream my-upstream '{"uri":["amqp://admin:admin@192.168.128.72/test2","amqp://admin:admin@192.168.128.71/test2","amqp://admin:admin@192.168.127.112/test2"],"expires":360000}'
```

```bash
abbitmqctl set_parameter federation-upstream my-upstream '{"uri":"amqp://admin:admin@192.168.127.112","expires":360000}'
```

**Comment by michaelklishin** [comment, marked as answer]
Thanks to @acogoluegnes, as of 5.2.0 there will be a getter for runtime parameters, in fact, two. They will each return a `List<RuntimeParameter<Object>>`, one with virtual host-specific parameters and the other with global parameters.

**Comment by acogoluegnes** [comment]
@stardustman Can you [test 5.2.0-SNAPSHOT](https://github.com/rabbitmq/hop#snapshots) to see if it's OK for you? You should get a `Map` for the value of your parameters. Thanks.

**Reply by stardustman** [reply]
ok, Thanks, I will try it.
